fmriprep/reports/core.py

```diff
@@ -22,100 +22,45 @@
 
-# This patch is intended to permit fMRIPrep 20.2.0 LTS to use the YODA-style
-# derivatives directory. Ideally, we will remove this in 20.3.x and use an
-# updated niworkflows.
 
-
-class Report(_Report):
-    def _load_config(self, config):
-        from yaml import safe_load as load
-
-        settings = load(config.read_text())
-        self.packagename = self.packagename or settings.get("package", None)
-
-        # Removed from here: Appending self.packagename to self.root and self.out_dir
-        # In this version, pass reportlets_dir and out_dir with fmriprep in the path.
-
-        if self.subject_id is not None:
-            self.root = self.root / f"sub-{self.subject_id}"
-
-        if "template_path" in settings:
-            self.template_path = config.parent / settings["template_path"]
-
-        self.index(settings["sections"])
-
-
-#
-# The following are the interface used directly by fMRIPrep
-#
-
-
-def run_reports(
-    out_dir,
-    subject_label,
-    run_uuid,
-    config=None,
-    reportlets_dir=None,
-    packagename=None,
-):
-    """
-    Run the reports.
-
-    .. testsetup::
-
-    >>> copytree_or_skip("data/tests/work", testdir)
-    >>> (testdir / 'fmriprep').mkdir(parents=True, exist_ok=True)
-
-    .. doctest::
-
-    >>> run_reports(testdir / 'out', '01', 'madeoutuuid', packagename='fmriprep',
-    ...             reportlets_dir=testdir / 'work' / 'reportlets' / 'fmriprep')
-    0
-
-    """
-    return Report(
-        out_dir,
-        run_uuid,
-        config=config,
-        subject_id=subject_label,
-        packagename=packagename,
-        reportlets_dir=reportlets_dir,
-    ).generate_report()
-
-
-def generate_reports(
-    subject_list, output_dir, run_uuid, config=None, work_dir=None, packagename=None
-):
-    """Execute run_reports on a list of subjects."""
+def generate_reports(subject_list, output_dir, run_uuid, config=None, work_dir=None):
+    """Generate reports for a list of subjects."""
     reportlets_dir = None
     if work_dir is not None:
         reportlets_dir = Path(work_dir) / "reportlets"
-    report_errors = [
-        run_reports(
+
+    error_list = ""
+    for subject_label in subject_list:
+        entities = {}
+        entities["subject"] = subject_label
+
+        robj = Report(
             output_dir,
-            subject_label,
             run_uuid,
-            config=config,
-            packagename=packagename,
+            bootstrap_file=config,
             reportlets_dir=reportlets_dir,
+            plugins=None,
+            plugin_meta=None,
+            metadata=None,
+            **entities,
         )
-        for subject_label in subject_list
-    ]
 
-    errno = sum(report_errors)
+        # Count nbr of subject for which report generation failed
+        errno = 0
+        try:
+            robj.generate_report()
+        except:
+            errno += 1
+            error_list = error_list + f"{subject_label}, "
+
```

Comment:
See commient about the config singleton (it is an unrelated object at this point). Let's just return here.

```suggestion

    return errors
```

IMPORTANT - please note that returning thing will require you to remove the remainder of the code in this function -- I can't do that with the suggestion tool, apparently.